hibernate-core/src/main/java/org/hibernate/resource/beans/spi/ManagedBeanRegistry.java

```diff
@@ -24,15 +24,48 @@ public interface ManagedBeanRegistry extends Service {
 	<T> ManagedBean<T> getBean(Class<T> beanClass);
 
 	/**
-	 * Get a bean reference by name and contract.
+	 * Form of {@link #getBean(Class)} allowing to indicate whether
+	 * it is required to use CDI if it is available
+	 *
+	 * @implNote Defaulted for backwards compatibility
 	 */
-	<T> ManagedBean<T> getBean(String beanName, Class<T> beanContract);
+	default <T> ManagedBean<T> getBean(Class<T> beanClass, boolean cdiRequiredIfAvailable) {
```

Comment:
You might want to use an enum instead of a boolean, both for more explicit code on call sites and to be free to add other behavior later if need be?